pkgdb/include/flox/lock-flake-installable.hh

```diff
@@ -0,0 +1,138 @@
+/* ========================================================================== *
+ *
+ * @file flox/lock-flake-installable.hh
+ *
+ * @brief Executable command helper and `flox::lockFlakeInstallable`.
+ *
+ *
+ * -------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <filesystem>
+
+#include <argparse/argparse.hpp>
+
+#include "flox/core/command.hh"
+#include "flox/core/nix-state.hh"
+
+
+/* -------------------------------------------------------------------------- */
+
+namespace flox {
+
+/* -------------------------------------------------------------------------- */
+
+/** @brief Lock a flake installable for flox */
+class LockFlakeInstallableCommand : flox::NixState
+{
+
+private:
+
+  command::VerboseParser parser;
+
+  std::string installable;
+  std::string system = nix::settings.thisSystem.get();
+
+public:
+
+  LockFlakeInstallableCommand();
+
+  [[nodiscard]] command::VerboseParser &
+  getParser()
+  {
+    return this->parser;
+  }
+
+  /**
+   * @brief Execute the `lock` routine.
+   * @return `EXIT_SUCCESS` or `EXIT_FAILURE`.
+   */
+  int
+  run();
+};
+
+
+/* -------------------------------------------------------------------------- */
+
+struct LockedInstallable
+{
+  std::string                             lockedUrl;
```

Comment:
nonblocking note: I realized when reading through tests that there's not a great way for the CLI to disallow `path` flakes. We'll probably need to add `type` to `LockedInstallable` if we want to error in Rust or add a check for that in C++ (except for when we're running tests 🤔 )